api/app/crud/sample.py

```diff
@@ -77,12 +79,51 @@ async def get_samples_summary(
     pipeline.append(
         {
             "$addFields": {
-                "typing_result": {
-                    "$filter": {
-                        "input": "$typing_result",
-                        "as": "res",
-                        "cond": {"$eq": ["$$res.type", "mlst"]},
+                "mlst": {
```

Comment:
The adding of bot `mlst` and `stx` fields via `$addFields` is not actually necessary because of the optional projections (line 145 & 147), or have I misunderstood something?

Reply:
I think you need to keep `$addFields`. The projections only specify which fileds in the output document that should be sent back, not how they (in this case) are created.